### ISSUE TITLE
Potential panic in `Signature::try_from`

### ISSUE THREAD

**Issue by nbaztec** [RustCrypto/signatures#563]
**What**
Using `Signature::try_from` can panic in a specific case, due to the interactions between rust's auto-implementations and user oversight. 

**Expected Behavior**
`try_from` should either return an `Ok` or `Error`

**Actual Behavior**
Given the following code, if the user makes the mistake of passing the bytes as value `[u8; 64]` instead of a reference `&[u8]`, `try_from` will accept that input and potentially panic at runtime for invalid input.

```rust
let sig = [255u8; 64];
if let Ok(s) = Signature::try_from(sig) {  // should've been `sig.as_ref()`
  ...
}
```
```
thread 'main' panicked at 'invalid signature: signature::Error { source: None }', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/ed25519-1.5.2/src/lib.rs:331:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Explanation**
This is because the `From<[u8; 64]>` albeit deprecated, is still implemented. Even though `TryFrom<&[u8]>` is the only impl available, Rust [auto-impls](https://doc.rust-lang.org/stable/src/core/convert/mod.rs.html#598) `TryFrom<[u8; 64]>` via `From<u8; 64>`.

This is quite significant as a single error can give the user the impression that they'll get a `Result`, whereas the codepath can potentially panic instead. 

Removing `From<[u8; 64]>` implementation should fix this behavior.

**Comment by tarcieri** [comment]
I assume this is in regard to the `ed25519 crate? There are several crates in this repo.

Please see the comments on the conversion:

https://github.com/RustCrypto/signatures/blob/1eec6b4/ed25519/src/lib.rs#L365-L376

```
/// DEPRECATED: use `TryFrom<&[u8]>` instead.
///
/// # Warning
///
/// This conversion will panic if a signature is invalid.
// TODO(tarcieri): remove this in the next breaking release
```

As you can see, all of this is already clearly documented.

Fortunately, as it were, we literally merged the first PR to move towards an `ed25519` v2.0 release yesterday: #562 

So yes, that conversion can now be removed.

**Comment by nbaztec** [comment]
Thanks, yes this was with regards to `ed25519`. The issue came up due to `try_from` implicitly using `from`, which is easy to miss with the current design. Thanks for removing it!
